testenv/utils/ssh.go

```diff
@@ -0,0 +1,171 @@
+// Copyright © 2024 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"path/filepath"
+
+	"golang.org/x/crypto/ssh"
+)
+
+type SSHKeyPair struct {
+	PublicKeyFile  string
+	PrivateKeyFile string
+	Temporary      bool
+}
```

Comment:
```suggestion
type SSHKeyPair struct {
	PublicKey  []byte
	PrivateKey []byte
	Temporary bool
}

func (k *SSHKeyPair) Export(privKeyFile, pubKeyFile string) error {}
func (k *SSHKeyPair) Load(privKeyFile, pubKeyFile string) error {}
```
I'd make the `SSHKeyPair` simple container which supports key-pairs to be loaded from/exported to the filesystem.